src/depends.py

```diff
@@ -1,52 +1,199 @@
-#! python
+"""
+Utility functions to check the availability of dependencies
+and suggest how it may be installed
+"""
 
 import sys
-import os
-import pyelliptic.openssl
 
 # Only really old versions of Python don't have sys.hexversion. We don't
 # support them. The logging module was introduced in Python 2.3
 if not hasattr(sys, 'hexversion') or sys.hexversion < 0x20300F0:
-    sys.stdout.write('Python version: ' + sys.version)
-    sys.stdout.write('PyBitmessage requires Python 2.7.3 or greater (but not Python 3)')
-    sys.exit()
+    sys.exit(
+        'Python version: %s\n'
+        'PyBitmessage requires Python 2.7.3 or greater (but not Python 3)'
+        % sys.version
+    )
 
-# We can now use logging so set up a simple configuration
 import logging
+import os
+from importlib import import_module
+
+# We can now use logging so set up a simple configuration
 formatter = logging.Formatter(
     '%(levelname)s: %(message)s'
 )
 handler = logging.StreamHandler(sys.stdout)
 handler.setFormatter(formatter)
-logger = logging.getLogger(__name__)
+logger = logging.getLogger('both')
 logger.addHandler(handler)
 logger.setLevel(logging.ERROR)
 
+PACKAGE_MANAGER = {
```

Comment:
Why can't we have all these imports in a json and import where they required.

Reply:
This is a special case, this file is for verifying what modules are available in the system so that it can take compensating measures or fail gracefully while explaining to the user what he has to do.